src/secret/ledger_nano.rs

```diff
@@ -95,160 +101,209 @@ impl SecretManage for LedgerSecretManager {
     }
 }
 
+// needs_blindsigning
+// the Ledger Nano S(+)/X app can present the user a detailed view of the transaction before it
+// is signed but only with BasicOutputs without extra-features. For everything else,
+// blindsigning is needed. This method finds out if we have to switch to blindsigning mode.
+fn needs_blindsigning(prepared_transaction: &PreparedTransactionData) -> bool {
+    match &prepared_transaction.essence {
+        bee_block::payload::transaction::TransactionEssence::Regular(essence) => {
+            for output in essence.outputs().iter() {
+                // only basic outputs allowed
+                if let bee_block::output::Output::Basic(s) = output {
+                    // no native tokens
+                    // only one address unlock
+                    // no features
+                    if let ([], [bee_block::output::UnlockCondition::Address(_)], []) = (
+                        s.native_tokens().as_ref(),
+                        s.unlock_conditions().as_ref(),
+                        s.features().as_ref(),
+                    ) {
+                        // all fine, continue with next output
+                        continue;
+                    }
+                }
+                // not fine, return
+                return true;
+            }
+        }
+    }
+    false
```

Comment:
We still need to check the maximum amount of inputs/outputs based on the used ledger nano device